### PR TITLE
Merge ironic and ironic-inspector repositories, single Dockerfile

### PR DESCRIPTION
**Pull request by elfosardo**
This change moves Dockerfile, configuration and scripts from the
ironic-inspector container image repository to the ironic one.

This approach provides one Dockerfile that builds an image that contains
all the services necessary to run both ironic and ironic-inspector.
The scripts and configurations are kept separated for the time being as
some of them conflict with each other (e.g. runhttpd) and are being
renamed to avoid overwriting during the image building process.

Those changes will need to be reflected in CI as well.

This change is mutually exclusive with #252

### CODE REVIEW

File: Dockerfile
Comment (dtantsur) [inline Dockerfile:80]: Should be removed

File: ironic-inspector-scripts/runhealthcheck
Comment (dtantsur) [inline ironic-inspector-scripts/runhealthcheck:1]: I think we got rid of these custom healthchecks already, let's not carry them over here.

File: ironic-inspector-scripts/runhttpd-inspector
Comment (dtantsur) [inline ironic-inspector-scripts/runhttpd-inspector:1]: One of the key points of this work is to not run two httpd instances, only one to Rule Them ALL!

**Review by dtantsur — COMMENTED** [review]
I like this path, just a couple of comments inline.

File: Dockerfile
Comment (elfosardo) [reply Dockerfile:80]: yep, this needs change in BMO at least as we're still relying on it
https://github.com/metal3-io/baremetal-operator/pull/851

File: ironic-inspector-scripts/runhealthcheck
Comment (elfosardo) [reply ironic-inspector-scripts/runhealthcheck:1]: sounds good to me! :)

File: ironic-inspector-scripts/runhttpd-inspector
Comment (elfosardo) [reply ironic-inspector-scripts/runhttpd-inspector:1]: yes, I changed the name here because when we run httpd for ironic, we can run runhttpd normally, while when running inspector we run runhttpd-inspector, configurations are quite different
I can have a look at merging the two scripts in one, but in any case we won't end up running two instances, there will be only one depending on which service we're running

Comment (elfosardo) [reply ironic-inspector-scripts/runhttpd-inspector:1]: still need to change this

**Comment by fmuyassarov** [comment]
I assume we will have a single image (named Ironic..)? If yes, then we need to 

1. clean up ironic-inspector image references from Metal3-dev-env, some examples: [ref1](https://github.com/metal3-io/metal3-dev-env/blob/40e3f79b78d4f317e117b98731230a7c929f5307/vm-setup/roles/v1aX_integration_test/tasks/move.yml#L13), [ref2](https://github.com/metal3-io/metal3-dev-env/blob/9886555e8abb71fc58d676ba77553567e5aa8163/lib/common.sh#L159), etc.
2. remove image building process from [quay](https://quay.io/repository/metal3-io/ironic-inspector) (this can be done after this PR goes in I think)

**Comment by elfosardo** [comment]
> I assume we will have a single image (named Ironic..)? If yes, then we need to
> 
>     1. clean up ironic-inspector image references from Metal3-dev-env, some examples: [ref1](https://github.com/metal3-io/metal3-dev-env/blob/40e3f79b78d4f317e117b98731230a7c929f5307/vm-setup/roles/v1aX_integration_test/tasks/move.yml#L13), [ref2](https://github.com/metal3-io/metal3-dev-env/blob/9886555e8abb71fc58d676ba77553567e5aa8163/lib/common.sh#L159), etc.
> 
>     2. remove image building process from [quay](https://quay.io/repository/metal3-io/ironic-inspector) (this can be done after this PR goes in I think)

Thanks @fmuyassarov !
that's correct, we'll have a single image that contains all needed for ironic and ironic-inspector
I was also looking at metal3-dev-env, but I think that also needs to be done after this PR merges?
I would also like to wait for https://github.com/metal3-io/baremetal-operator/pull/851 and https://github.com/metal3-io/ironic-inspector-image/pull/81 to merge first

**Comment by fmuyassarov** [comment]
> > I assume we will have a single image (named Ironic..)? If yes, then we need to
> > ```
> > 1. clean up ironic-inspector image references from Metal3-dev-env, some examples: [ref1](https://github.com/metal3-io/metal3-dev-env/blob/40e3f79b78d4f317e117b98731230a7c929f5307/vm-setup/roles/v1aX_integration_test/tasks/move.yml#L13), [ref2](https://github.com/metal3-io/metal3-dev-env/blob/9886555e8abb71fc58d676ba77553567e5aa8163/lib/common.sh#L159), etc.
> > 
> > 2. remove image building process from [quay](https://quay.io/repository/metal3-io/ironic-inspector) (this can be done after this PR goes in I think)
> > ```
> 
> Thanks @fmuyassarov !
> that's correct, we'll have a single image that contains all needed for ironic and ironic-inspector
> I was also looking at metal3-dev-env, but I think that also needs to be done after this PR merges?
> I would also like to wait for [metal3-io/baremetal-operator#851](https://github.com/metal3-io/baremetal-operator/pull/851) and [metal3-io/ironic-inspector-image#81](https://github.com/metal3-io/ironic-inspector-image/pull/81) to merge first

Yeah, I think metal3-dev-env changes can be done later too, because we will still have image for ironic-inspector in quay, so that won't really break the CI.

**Comment by fmuyassarov** [comment]
I forgot about BMO, [ref1](https://github.com/metal3-io/baremetal-operator/blob/993f2892d42544fdc168afa744c99c885f746b44/ironic-deployment/ironic/ironic.yaml#L88), [ref2](https://github.com/metal3-io/baremetal-operator/blob/488efeda1a816cd6b10cb32e63afdd00b355d6d0/tools/run_local_ironic.sh#L8), etc.

**Review by dtantsur — APPROVED** [review]
\o/

**Comment by elfosardo** [comment]
@maelk @fmuyassarov @hardys can you please have another look at this? Thanks!

**Review by fmuyassarov — COMMENTED** [review]
Thanks, this LGTM. We will probably need to force merge this one too, considering tide isn't in place at this moment.

**Comment by elfosardo** [comment]
@fmuyassarov thanks! As clarified on our chat, I'm totally ok waiting for tide to be back before we merge this.
We can re-evaluate the status during the week or early next week

**Comment by dtantsur** [comment]
/lgtm

Tide is back?

**Comment by fmuyassarov** [comment]
> /lgtm
> 
> Tide is back?

It was for a while when I was testing. Still there are some minor things to setup :) but hopefully by Monday will be up.

**Comment by fmuyassarov** [comment]
@elfosardo @dtantsur Prow is back on track. Probably `/lgtm `will be enough to get this PR in.

**Comment by dtantsur** [comment]
/lgtm

pretty please

**Pull request merged**
